app/src/main/java/org/hackillinois/android/App.kt

```diff
@@ -0,0 +1,58 @@
+package org.hackillinois.android
+
+import android.app.Application
+import android.arch.persistence.room.Room
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import org.hackillinois.android.database.Database
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class App : Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+        database = Room.databaseBuilder(applicationContext, Database::class.java, "local-db")
+                .fallbackToDestructiveMigration()
+                .build()
+    }
+
+    companion object {
+        private var apiInitialized = false
+        private lateinit var apiInternal: API
+
+        lateinit var database: Database
+            private set
+
+        fun getAPI(token: String? = null): API {
+            if (token == null) {
+                return if (apiInitialized) {
```

Comment:
`return if` is a little confusing in my opinion; you could throw this all on one line like a ternary